### ISSUE TITLE
Questions about the paper

### ISSUE THREAD

**Issue by DBraun** [SonyCSLParis/music2latent#4]
Thanks for this great research.

It seems possible to write the training code based on the paper, but I think there are some loose ends.

In equation 9, the "Pseudo-Huber" loss function, is `c` .03, or does it depend on the spectrogram size?

For equation 14, when sampling $t_{i+1}$, could you describe the lognormal approach more? I don't understand how it relates to the ICT paper, since that uses $P_{mean}$ and $P_{std}$ with `torch.special.erf` but with a **discrete** sigma schedule.

**Comment by marcoppasini** [comment]
Hey there!

Regarding the huber loss function, as specified in the paper I adopt the same heuristic as in https://arxiv.org/abs/2310.14189: "We suggest that c should scale linearly with ∥x - y∥2, and propose a heuristic of c = 0.00054*sqrt(d) for images with d dimensions", where I treat spectrograms as images.

Regarding the lognormal, I simply approximate it by using an arbitrary very high number of discrete steps.

**Comment by DBraun** [comment]
Thanks! I think I'll use linspace to make the domain of the lognormal [.0001, 1], use p_mean and p_std, then normalize to get a PDF, and sample.
